packages/quantic/force-app/main/default/lwc/quanticSearchBox/quanticSearchBox.js

```diff
@@ -126,13 +166,35 @@ export default class QuanticSearchBox extends LightningElement {
     }
     this.state = this.searchBox?.state;
     this.suggestions =
-      this.state?.suggestions?.map((s, index) => ({
+      this.state?.suggestions?.map((suggestion, index) => ({
         key: index,
-        rawValue: s.rawValue,
-        value: s.highlightedValue,
+        rawValue: suggestion.rawValue,
+        value: suggestion.highlightedValue,
       })) ?? [];
   }
 
+  updateRecentQueriesListState() {
+    if (this.recentQueriesList.state?.queries) {
+      this.recentQueries = this.recentQueriesList.state.queries;
+      setItemInLocalStorage(
+        this.localStorageKey,
+        this.recentQueriesList.state.queries
+      );
```

Comment:
For your first question: `This completely overrides the current existing recent queries in the Local Storage if some exists, is that expected`

No this will no override the existing queries, in other words no recent query will be lost because the headless state contains the same piece of information that can be found in the local storage because in the initialization we do the folowing:
```
      this.recentQueriesList = this.headless.buildRecentQueriesList(engine, {
        initialState: {
          queries: getItemFromLocalStorage(this.localStorageKey) ?? [],
        },
       ...
```

Reply:
For your second question about the Standalone search box and how recent queries are handled with this component:

for now now that is what is supported:

- If you perform a query with  standalone searchbox, it will be added to the recent queries and it will be displayed as a recent query in the search box of the search page:

https://github.com/coveo/ui-kit/assets/86681870/221c9706-d960-4028-a907-84e571f78451


Otherwise, the standalone search box will NOT display  recent queries.

To make this happen, it's fairly complicated technically, we will have to always  keep the state of of the Standalone engine similar to the state of the search engine used in the search page. We can do it but it will not be pretty it will over complicate the QuanticSearchBox component and the QuanticStandaloneSearchBox component.

So in my opinion it will not be worth it to support that, especially that naturally the standalone search box does not need recent queries as usually it is used to perform the first query, then it redirects to the search way where you will  start seeing the recent queries as expected in the search box.  

We cam evaluate of course and decide to add it, but in all cases I would consider this in another iteration as the solution to accomplish this will be complicated/hackish when I imagine it x) and for sure it will come with some corner cases that we will need to pay attention to so it will need to be very heavily tested, that's why I don't see it in this PR.

Reply:
Well what I'm saying is that it might be a better idea to rely on the localStorage more than to rely on the state of Headless for that exact reason.

For example too, I know customers have multiple searchboxes. One on the main page, one inline in the search page as well as one in the header too only visible outside of the home page and search page.

I believe it would make more sense to have the recent queries all show up in these searchboxes no?

Already that the recent queries are not available on the standalone searchbox outside of the search page is a bit worrying. It will generate questions.

I wonder how Atomic does it with a standalone searchbox? Do you know?

Reply:
I'm wondering if we shouldn't instead have some kind of `uniqueList` of recent queries in localStorage, no matter where they are coming from.

So that any standalone searchbox can also display them.

They would indeed be updated based on this `updateRecentQueriesListState` but the `localStorageKey` could be shared across all Coveo Quantic components in your community for example.

Reply:
As discussed previously, like you mentioned the only way to support recent queries in the standalone search box to have the `localStorageKey` shared across all Coveo Quantic components and that's how Atomic does it.

In our case, I would challenge this decision because we are likely to have multiple engines running in the same time, in the HIP for example with multiple cases, it can become messy if qll recent queries are shared.

so I think the best thing to do now to separate this concern.
We start by not support recent queries in the standalone search box, and we evaluate the needs/risks of supporting it separately in a second iteration.